end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java

```diff
@@ -84,28 +80,40 @@ public void startCalls(ResultTable results) throws InterruptedException, IOExcep
         switch (args.method) {
           case METHOD_READ:
             for (int i = 0; i < args.threads; i++) {
-              Runnable task =
-                  () -> {
-                    try {
-                      makeMediaRequest(results);
-                    } catch (IOException e) {
-                      e.printStackTrace();
-                    }
-                  };
+              int finalI = i;
```

Comment:
Could you explain what is the meaning of this name `finalI`? It seems this local variable is not necessary.

Reply:
This is to capture the value of `i`. https://stackoverflow.com/questions/54340101/enhanced-for-loop-and-lambda-expressions